### PR TITLE
feat: change launch warning to launch error

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
We want to block users from submitting experiments that will never get scheduled. However, this renders a lot of the LaunchWarning mechanisms useless.

We should discuss whether we want to change the LaunchWarning interfaces or keep launch warning and move the cancellation mechanism deeper in the scheduler.

## Test Plan
Create an experiment with more slots than the cluster has. The experiment should not be created and the command will error out.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by ioga** [comment]
stock warning -> error change will be a risky, breaking change. this warning was wrong on a few occasions. we should seriously consider making it optional.

File: docs/reference/reference-deploy/config/master-config-reference.rst [hunk 1]
```diff
@@ -113,6 +113,9 @@ The master supports the following configuration settings:
       ``/var/cache/determined``. Note that the master would break on startup if it does not have
       access to create this default directory.
 
+-  ``launch_error`` (optional): Specify whether to refuse an experiment or task if the slots
+   requested exceeds the cluster capacity. Defaults to ``false`` and shows a warning instead.
```
Comment (ioga) [inline docs/reference/reference-deploy/config/master-config-reference.rst:117]: let's default to true, i.e. error.  
- if it turns out to be broken for some users, they can opt out
- if it works 100% of the time and everyone loves it, we remove this option

**Review by ioga — APPROVED** [review]

File: docs/reference/reference-deploy/config/master-config-reference.rst [hunk 2]
```diff
@@ -113,6 +113,10 @@ The master supports the following configuration settings:
       ``/var/cache/determined``. Note that the master would break on startup if it does not have
       access to create this default directory.
 
+-  ``launch_error`` (optional): Specify whether to refuse an experiment or task if the slots
+   requested exceeds the cluster capacity. If ``false``, only a warning will be returned. Defaults
+   to ``true``.
+
```
Comment (tara-det-ai) [inline docs/reference/reference-deploy/config/master-config-reference.rst:119]: -  ``launch_error`` (optional): Specifies whether to refuse an experiment or task if the slots
   requested exceeds the cluster capacity. If set to ``false``, only a warning is returned. The default value is ``true``.

also, is it "the cluster capacity" or the "cluster's capacity"?

**Review by tara-det-ai — APPROVED** [review]
suggested edit https://github.com/determined-ai/determined/pull/6412/files#r1164420209

File: docs/reference/reference-deploy/config/master-config-reference.rst (hunk 2 above)
Comment (eecsliu) [reply docs/reference/reference-deploy/config/master-config-reference.rst:119]: I think it should be the "cluster's capacity"

**Review by erikwilson — APPROVED** [review]

**Pull request merged**